src/components/Form.js

```diff
@@ -3,77 +3,153 @@ import TextInput from './TextInput';
-  propsArray[2] = { ...propsArray[2], handleDatepicker };
-  propsArray[3] = { ...propsArray[3], handleAutocomplete };
-  propsArray[4] = { ...propsArray[4], handleDatepicker };
-  return (
-    <form>
-      <TextInput {...propsArray[0]} />
-      <TextInput {...propsArray[1]} />
-      <Datepicker {...propsArray[2]} />
-      <Autocomplete {...propsArray[3]} />
-      <Datepicker {...propsArray[4]} />
-      <TextInput {...propsArray[5]} />
-      <RadioButton {...propsArray[6]} />
-      <RadioButton {...propsArray[7]} />
-      <RadioButton {...propsArray[8]} />
-    </form>
-  );
-};
+class Form extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      eventName: '',
+      eventWebsite: '',
+      eventDate: '',
+      eventLocation: '',
+      submissionDueDate: '',
+      submissionWebsite: '',
+      isCompensated: '',
+      hasCodeOfConduct: '',
+      hasDiversityScholarships: '',
+      contactName: '',
+      contactEmail: ''
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleAutocomplete = this.handleAutocomplete.bind(this);
+    this.handleDatepicker = this.handleDatepicker.bind(this);
+    this.showState = this.showState.bind(this);
+  }
+  showState() {
+    alert(this.state);
+  }
+  handleChange(e) {
+    this.setState({ [e.target.name]: e.target.value });
+  }
 
+  handleAutocomplete(city) {
+    this.setState({ eventLocation: city });
+  }
+  handleDatepicker(selectedDate, dateName) {
+    this.setState({ [dateName]: selectedDate });
+  }
+  makePropsArray() {
+    let propsArray = [
+      {
+        title: 'Event Name',
+        name: 'eventName',
+        inputType: 'text'
+      },
+      {
+        title: 'Event Website',
+        name: 'eventWebsite',
+        inputType: 'text'
+      },
+      {
+        title: 'Event Date',
+        name: 'eventDate'
+      },
+      {
+        title: 'Event Location',
+        name: 'eventLocation'
+      },
+      {
+        title: 'Submission Due Date',
+        name: 'submissionDueDate'
+      },
+      {
+        title: 'Submission Website',
+        name: 'submissionWebsite'
+      },
+      {
+        name: 'isCompensated',
+        question: 'Are all speakers compensated at your event?',
+        options: ['Yes', 'No']
+      },
+      {
+        name: 'hasCodeOfConduct',
+        question: 'Does your event have a publicly visible code of conduct?',
+        options: ['Yes', 'No']
+      },
+      {
+        name: 'hasDiversityScholarships',
+        question: 'Does your event provide diversity scholarships?',
+        options: ['Yes', 'No']
+      },
+      {
+        name: 'contactName',
+        title: 'Contact Name'
+      },
+      {
+        name: 'contactEmail',
+        title: 'Contact Email'
+      }
+    ];
+    return propsArray;
+  }
+  render() {
+    let propsArray = this.makePropsArray();
+    propsArray = propsArray.map(current => ({
+      ...current,
+      handleChange: this.handleChange
```

Comment:
The `handleChange`, `handleDatepicker`, etc methods can be passed directly into the components that need them as props.